base/broadcast.jl

```diff
@@ -338,46 +338,46 @@ for (f, scalarf, bitf, bitfbody) in ((:.==, :(==), :biteq , :(~a $ b)),
     end
 end
 
+function bitcache(op, A, B, refA, refB, l::Int, ind::Int, C::Vector{Bool})
+    left = l - ind + 1
+    @inbounds begin
+        for j = 1:min(bitcache_size, left)
+            C[j] = (op)(refA(A, ind), refB(B, ind))
+            ind += 1
+        end
+        C[left+1:bitcache_size] = false
+    end
+    return ind
+end
+
 # note: the following are not broadcasting, but need to be defined here to avoid
 # ambiguity warnings
 
-for (f, cachef, scalarf) in ((:.==, :bitcache_eq , :(==)),
-                             (:.< , :bitcache_lt , :<   ),
-                             (:.!=, :bitcache_neq, :!=  ),
-                             (:.<=, :bitcache_le , :<=  ))
-    @eval ($cachef)(A::AbstractArray, B::AbstractArray, l::Int, ind::Int, C::Vector{Bool}) = 0
-    for (sigA, sigB, expA, expB, shape) in ((:Any, :AbstractArray,
-                                             :A, :(B[ind]),
-                                             :(size(B))),
-                                            (:AbstractArray, :Any,
-                                             :(A[ind]), :B,
-                                             :(size(A))))
+for (f, scalarf) in ((:.==, ==),
```

Comment:
I don't think it's necessary with jb/functions. But neither would it hurt, so I'll add it back.